src/FSharpPlus/Control/Traversable.fs

```diff
@@ -95,8 +95,15 @@ type Traverse =
         | Choice2Of2 e -> Return.Invoke (Choice<'U,'Error>.Choice2Of2 e)
 
     static member inline Traverse (t:list<_>   ,f , [<Optional>]_output: 'R, [<Optional>]_impl: Traverse) : 'R =
-       let cons_f x ys = Map.Invoke List.cons (f x) <*> ys
-       List.foldBack cons_f t (result [])
+        let rec loop acc = function
+            | [] -> acc
+            | x::xs -> let v = f x
+                       if v |> IsLeftZero.Invoke |> not 
```

Comment:
Using `IsLeftZero` here changes the signature and hence the overload resolution logic.
I know it would make it more optimal but optimization should be a separate PR.